driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -334,57 +353,57 @@ private MongoTimeoutException createTimeoutException(final Timeout timeout) {
     }
 
 
+    /**
+     * Is package-access for the purpose of testing and must not be used for any other purpose outside of this class.
+     */
     ConcurrentPool<UsageTrackingInternalConnection> getPool() {
         return pool;
     }
 
-    private Runnable createMaintenanceTask() {
-        Runnable newMaintenanceTask = null;
-        if (shouldPrune() || shouldEnsureMinSize()) {
-            newMaintenanceTask = new Runnable() {
-                @Override
-                public synchronized void run() {
+    /**
+     * Is package-access for the purpose of testing and must not be used for any other purpose outside of this class.
+     * <p>
+     * Synchronously prune idle connections and ensure the minimum pool size.
+     */
+    void doMaintenance() {
+        Predicate<RuntimeException> silentlyComplete = e ->
+                e instanceof MongoInterruptedException || e instanceof MongoTimeoutException
+                        || e instanceof MongoConnectionPoolClearedException || ConcurrentPool.isPoolClosedException(e);
+        try {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(format("Pruning pooled connections to %s", serverId.getAddress()));
+            }
+            pool.prune();
```

Comment:
Yes, it would. Originally he code looked like
```java
int curGeneration = generation.get();
if (shouldPrune() || curGeneration > lastPrunedGeneration.get()) {
    pool.prune();
}
lastPrunedGeneration.set(curGeneration);
```
The LB PR changed it to
```java
if (shouldPrune()) {
    pool.prune();
}
```
Because the `lastPrunedGeneration` optimization was not sensible anymore for LB pools. We discussed it in Slack and decided to call `pool.prune()` each run of the background thread.